nifi-nar-bundles/nifi-riemann-bundle/nifi-riemann-processors/src/main/java/org/apache/nifi/processors/riemann/PutRiemann.java

```diff
@@ -0,0 +1,376 @@
+        context.yield();
+        throw new ProcessException(String.format("Unable to connect to Riemann [%s:%d] (%s)\n%s", host, port, transport, e.getMessage()));
+      }
+    }
+
+    if (customAttributes.size() == 0) {
+      for (Map.Entry<PropertyDescriptor, String> property : context.getProperties().entrySet()) {
+        // only custom defined properties
+        if (!getSupportedPropertyDescriptors().contains(property.getKey())) {
+          customAttributes.add(property.getKey());
+        }
+      }
+    }
+  }
+
+
+  @Override
+  public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+    // Check if the client is currently connected, as a previous trigger could have detected a failure
+    // in the connection.
+    if (riemannClient == null || !riemannClient.isConnected()) {
+      // clean up the client and attempt to re-initialize the processor
+      cleanUpClient();
+      onScheduled(context);
+    }
+
+    List<FlowFile> incomingFlowFiles = session.get(batchSize);
+    List<FlowFile> successfulFlowFiles = new ArrayList<>(incomingFlowFiles.size());
+    List<Event> eventsQueue = new ArrayList<>(incomingFlowFiles.size());
+    for (FlowFile flowFile : incomingFlowFiles) {
+      try {
+        eventsQueue.add(FlowFileToEvent.fromAttributes(context, customAttributes, flowFile));
+        successfulFlowFiles.add(flowFile);
+      } catch (NumberFormatException e) {
+        getLogger().warn(e.getMessage());
+        session.transfer(flowFile, REL_FAILURE);
+      }
+    }
+    try {
+      if (transport == Transport.TCP) {
+        Proto.Msg returnMessage = riemannClient.sendEvents(eventsQueue).deref(writeTimeout, TimeUnit.MILLISECONDS);
+        if (returnMessage == null) {
+          context.yield();
+          throw new ProcessException("Timed out writing to Riemann!");
+        }
+      } else {
+        riemannClient.sendEvents(eventsQueue);
+      }
+      riemannClient.flush();
+    } catch (Exception e) {
+      context.yield();
+      throw new ProcessException("Failed writing to Riemann\n" + e.getMessage());
+    }
+    session.transfer(successfulFlowFiles, REL_SUCCESS);
+    session.commit();
+  }
+
+  /**
+   * Converts a FlowFile into a Riemann Protobuf Event
+   */
+  private static class FlowFileToEvent {
+    protected static Event fromAttributes(ProcessContext context, List<PropertyDescriptor> customProperties,
+                                          FlowFile flowFile) {
+      Event.Builder builder = Event.newBuilder();
+
+      PropertyValue service = context.getProperty(ATTR_SERVICE).evaluateAttributeExpressions(flowFile);
+      if (service.getValue() != null && !service.getValue().trim().equals("")) {
+        builder.setService(service.getValue());
+      }
+      PropertyValue description = context.getProperty(ATTR_DESCRIPTION).evaluateAttributeExpressions(flowFile);
+      if (description.getValue() != null && !description.getValue().trim().equals("")) {
```

Comment:
if you're willing to add a dependency on Apache Commons-Lang, these checks can be simplified via [StringUtils.isBlank](https://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/StringUtils.html#isBlank(java.lang.String))